### ISSUE TITLE
Roadmap  🚀

### ISSUE THREAD

**Issue by wazizian** [iutzeler/skwdro#53]
# General
---

### Docs
- [x] Add docs
- [x] Add examples
### Entropic solver
- [x] Fix entropic solver (presample, postsample) #50 (WIP - almost ready)
- [x] Tune $\sigma$, $\epsilon$ (closed by #77 )
- [x] Pytorch API (postsample) + easy combination of RobustLoss & solve_dual
- [x] #80
### Tests
- [x] Add generalization tests (WIP - almost ready)
- [x] Close to cvxopt for entropic

### Dependencies
- [x] Check the Python dependencies
- [x] Check non-Python stuff (hatch, git-lfs) 

### Deprecated
- [x] Deprecate Numpy

### Done
- [x] #61
- [x] Pretrain $\theta$ ($\rho = 0$ #46)
- [x] Seed #21 
- [x] Check power of $\rho$ - $\rho_{norm}$ & $\rho_{cost}$
- [x] #75
- [x] Initialize $\lambda$ #56
- [x] Change structure of `NormCost` to accept labels
- [x] Penalty

---

- [x] Tests:
   - [x] #82
   - [x] Generalization @wazizian 
   - [x] Study influence of `n_samples` @floffy-f 
- [x] #85
- [x] Adaptative step optimizer as default @wazizian #83 
- [x] Exemples: @iutzeler 
   - [x] #89
   - [ ] Simple sto architecture (RBM ? HMM ? ...) @floffy-f 
   - [ ] WILDS https://wilds.stanford.edu/ @iutzeler 
   - [ ] Kaggle hell @ jerome
- [x] Check GPU @floffy-f  
- [x] Plots generalization

---
- [ ] Features:
   - [ ] Handle multiple dimensions for features on torch side (high-prio) @floffy-f  #90
   - [ ] Composite expectation (low-prio)
   - [ ] Bootstrap for $\lambda_0$ (low-prio)
   - [ ] Regularization (low-prio)
   - [ ] Specific error handling (low-prio) @floffy-f 
   - [ ] Handle/verify padded sequences for rnns (very-low-prio) @floffy-f
